script/versions.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+"""
+Query the github API for the git tags of a project, and return a list of
+version tags for recent releases, or the default release.
+
+The default release is the most recent non-RC version.
+
+Recent is a list of unqiue major.minor versions, where each is the most
+recent version in the series.
+
+For example, if the list of versions is:
+
+    1.8.0-rc2
+    1.8.0-rc1
+    1.7.1
+    1.7.0
+    1.7.0-rc1
+    1.6.2
+    1.6.1
+
+`default` would return `1.7.1` and
+`recent -n 3` would return `1.8.0-rc2 1.7.1 1.6.2`
+"""
+from __future__ import print_function
+
+import argparse
+import itertools
+import operator
+from collections import namedtuple
+
+import requests
+
+
+GITHUB_API = 'https://api.github.com/repos'
+
+
+class Version(namedtuple('_Version', 'major minor patch rc')):
+
+    @classmethod
+    def parse(cls, version):
+        version = version.lstrip('v')
+        version, _, rc = version.partition('-')
+        major, minor, patch = version.split('.', 3)
+        return cls(int(major), int(minor), int(patch), rc)
+
+    @property
+    def major_minor(self):
+        return self.major, self.minor
+
+    @property
+    def order(self):
+        return self[:3] + (None, self.rc) if self.rc else ('', )
+
+    def __str__(self):
+        rc = '-{}'.format(self.rc) if self.rc else ''
+        return '.'.join(map(str, self[:3])) + rc
+
+
+def group_versions(versions):
+    return list(
+        list(releases)
```

Comment:
I made a stupid mistake, I was thinking of sets while reading this, thinking if we used sets then that would give you the unique versions. Anyway. Disregard that bit.

What I'm trying to express is that to me, expanding this out isn't unnecessary code. This is harder to read. It reads well to you because you wrote it. As someone reading it to grok the solution and what problem it is solving, I'm sharing with you that this approach is harder to read in that it takes longer to figure out what's happening here. 

Balancing the line between very succinct code and over verbose code, we want somewhere in the middle. The rest of the codebase is written aiming for that somewhere in the middle. We don't have lots of functions that are inline like this, so I think it's fair to ask for it to be expanded out and made to be as readable as everything else we have.

Maybe worth pinging this @shin- and @aanand to see if it is just me who finds this harder to read. I appreciate my personal preferences and past experiences with code like this, means I'm not as much of a fan of denser functional coding style. 

I'm not blocking the PR on this, I am expressing my opinion.